thinc/layers/with_array.py

```diff
@@ -6,12 +6,11 @@
 from ..types import List2d
 
 
-ValT = TypeVar("ValT", bound=Array2d)
 SeqT = TypeVar("SeqT", bound=Union[Padded, Ragged, List2d, Array2d])
 
 
 @registry.layers("with_array.v1")
-def with_array(layer: Model[ValT, ValT], pad: int = 0) -> Model[SeqT, SeqT]:
```

Comment:
I think this change probably isn't quite right. The types are tricksy =/.

`Array2d` is a shorthand for `Union[Floats2d, Ints2d]`. If you ave `Model[Union[Floats2d, Ints2d], Union[Floats2d, Ints2d]`, that works out to:

```
Union[
  Model[Floats2d, Floats2d],
  Model[Ints2d, Ints2d],
  Model[Floats2d, Ints2d],
  Model[Ints2d, Floats2d]
]
```

That's not the right typing for the function, as only the first two are valid. That's why we use the type variable.

Reply:
Ah, yes, ofcourse! Thanks!